smallvil/src/main.rs

```diff
@@ -49,3 +49,15 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+fn detect_terminal() -> String {
+    use std::fs::read_link;
+
+    const SYMLINK: &str = "/usr/bin/x-terminal-emulator";
+
+    if let Ok(found) = read_link(SYMLINK) {
+        return read_link(&found).unwrap_or(found).to_string_lossy().to_string();
+    }
+
+    "/usr/bin/gnome-terminal".to_string()
+}
```

Comment:
Same goes for smallvil. While this is certainly nice to have (and I don't have any objections for this feature in general for anvil), I don't think smallvil is a good fit.